head_test.go

```diff
@@ -148,16 +127,24 @@ func TestNonDuplicateNaNDatapointsCausesAmendError(t *testing.T) {
 
 	app := hb.Appender()
 	_, err = app.Add(labels.Labels{}, 0, math.Float64frombits(0x7ff0000000000001))
-	if err != nil {
-		t.Fatalf("Failed to add sample: %s", err)
-	}
-	if err = app.Commit(); err != nil {
-		t.Fatalf("Unexpected error committing appender: %s", err)
-	}
+	require.NoError(t, err)
+	err = app.Commit()
+	require.NoError(t, err)
 
 	app = hb.Appender()
 	_, err = app.Add(labels.Labels{}, 0, math.Float64frombits(0x7ff0000000000002))
-	if err != ErrAmendSample {
-		t.Fatalf("Expected error amending sample, got: %s", err)
-	}
+	require.Equal(t, ErrAmendSample, err)
+
+	app = hb.Appender()
+	_, err = app.Add(labels.Labels{}, 0, math.NaN())
+	require.Equal(t, ErrAmendSample, err)
+
+	_, err = app.Add(labels.Labels{}, 1, math.NaN())
+	require.NoError(t, err)
+	err = app.Commit()
+	require.NoError(t, err)
+
+	app = hb.Appender()
+	_, err = app.Add(labels.Labels{}, 1, math.NaN())
+	require.Equal(t, ErrAmendSample, err)
```

Comment:
Hmm, but all this is assuming that we will add float comparisons later. I am pretty sure that won't happen. @fabxc WDYT?

If we have two tests doing the same thing, it is bound to confuse those w/o context and any minor refactor will merge the two tests. Will add another if you are sure, though.